BouaikIlham/my-budget
# My budget

## In this pull request :+1: 
 - [ ] A splash page with the app's name and links to the signup and log-in pages were created.
 - [ ] Sign up and log in to pages
 - [ ] The user can register in the app with their full name, email, and password (all mandatory).
 - [ ] The user can log into the app using email and password.
 - [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 - [ ] Home page (categories page)
 - [ ] When the user logs in, they are presented with the categories page.
 - [ ] For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
 - [ ] When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
 - [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
 - [ ] Transactions page
   -  At the top of the page, the user can see the name of the category and the total amount for it (sum of all of the amounts of the transactions in that category).
   - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
 - [ ] "Add a new category" page
 - [ ] The user fills out a form to create a new category, indicating their name and icon (both mandatory).
 - [ ] The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
 - [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
 - [ ] "Add a new transaction" page
 - [ ] The user fills out a form to create a new transaction with:
     - name (mandatory)
     - amount (mandatory)
     - categories (mandatory at least one)
- [ ] The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
   - Create tests for models user and category and operation 
   - Create integration tests for category and operation .

- [ ] You can check out the Video demo [here](https://www.loom.com/share/5d7e33d66e6348bd8b59b9595d670a96).

Thank you for your review in advance! nerd_face
